packages/devtools_app/lib/src/debugger/variables.dart

```diff
@@ -25,14 +25,19 @@ class Variables extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final controller = Provider.of<DebuggerController>(context);
-    if (controller.variables.value.isEmpty) return const SizedBox();
-    // TODO(kenz): preserve expanded state of tree on switching frames and
-    // on stepping.
-    return TreeView<DartObjectNode>(
-      dataRootsListenable: controller.variables,
-      dataDisplayProvider: (variable, onPressed) =>
-          displayProvider(context, variable, onPressed, controller),
-      onItemSelected: (variable) => onItemPressed(variable, controller),
+    return ValueListenableBuilder<List<DartObjectNode>>(
+      valueListenable: controller.variables,
+      builder: (context, variables, _) {
+        if (variables.isEmpty) return const SizedBox();
```

Comment:
we can actually remove the ValueListenableBuilder and the special case for variables.isEmpty. TreeView returns a SizedBox in its build method when there is an empty data set

Reply:
Oh nice! Good point